src/read_preference.ts

```diff
@@ -31,17 +33,25 @@ export interface ReadPreferenceOptions {
 }
 
 /** @public */
-export interface ReadPreferenceLikeOptions {
+export interface ReadPreferenceLikeOptions extends ReadPreferenceOptions {
   readPreference?:
     | ReadPreferenceLike
     | {
-        mode: ReadPreferenceMode;
-        preference: ReadPreferenceMode;
+        mode?: ReadPreferenceMode;
+        preference?: ReadPreferenceMode;
         tags: TagSet[];
         maxStalenessSeconds: number;
       };
 }
 
+/** @public */
+export interface ReadPreferenceFromOptions {
+  session?: ClientSession;
```

Comment:
I'm still confused as to why `session` makes the cut here? It doesn't seem relevant to a read preference

Reply:
Maybe I should make another type, but it comes from the `resolve` method that uses the session to determine the read pref from a transaction if there is one.

Reply:
ah! okay thanks, I forgot about that.